packages/client/src/modals/SuccessModal.js

```diff
@@ -0,0 +1,138 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Web3Context } from "contexts/Web3";
+import { useClipboard } from "../hooks";
+import { Copy, ArrowUp, OpenNewTab, Close } from "../components/Svg";
+import { getFlowscanUrlForTransaction } from "utils";
+import { ASSET_TYPES } from "../constants/enums";
+import { COIN_TYPE_TO_META, CONTRACT_NAME_TO_COIN_TYPE } from "../constants/maps";
+
+const SuccessModal = ({ data, type, txID, onClose, safeName, safeAddress, safeOwners }) => {
+    const { getNFTReference } = useContext(Web3Context);
+    const clipboard = useClipboard();
+
+    const [imageURI, setImageURI] = useState();
+
+    useEffect(() => {
+        if (type === ASSET_TYPES.NFT) {
+            const getImageURL = async () => {
+                const result = await getNFTReference(data.recipientAddr, data.nftID);
+                setImageURI(result);
+            };
+            getImageURL().catch(console.error);
+        }
+    }, [type])
+
+    function getNFTName(collectionId) {
+        return collectionId.split(".")[2];
+    }
+
+    function getTokenName(vaultId) {
+        const tokenName = vaultId.split(".")[2];
+        return COIN_TYPE_TO_META[CONTRACT_NAME_TO_COIN_TYPE[tokenName]].displayName;
+    }
+
+    function getTokenIcon(vaultId) {
+        const tokenName = vaultId.split(".")[2];
+        return COIN_TYPE_TO_META[CONTRACT_NAME_TO_COIN_TYPE[tokenName]].icon;
+    }
+
+    function getSafeOwnerName(address) {
+        const owner = safeOwners.filter(owner => owner.address === address);
+        return owner && owner[0].name;
+    }
+
+    return (
+        <div className="p-5 has-text-black has-text-left">
+            <div className="border-light-bottom columns is-vcentered is-multiline is-mobile">
+                <h2 className="pb-4 is-size-5 has-text-black column">Success</h2>
+                <a className="" onClick={onClose}>
+                    <Close className="mr-4" />
+                </a>
+            </div>
+
+            <div className="mt-4 p-5 success-modal-background">
+                <button className="button flex-1 is-info mb-1">
+                    Sent <ArrowUp className="ml-2 has-text-white" />
+                </button>
+                <div className="pl-4">
+                    {type === ASSET_TYPES.NFT &&
+                        <>
+                            <img className="columns is-vcentered is-multiline is-mobile mr-2 mt-2 success-modal-image" src={imageURI} />
```

Comment:
imageURI could be undefined, suggest conditional rendering this image checking imageURI first

Reply:
Fixed!